### PR TITLE
Add Spotify commands

### PR DESCRIPTION
**Pull request by gc**
getSpotifyToken Task: With the developers client ID and client secret, it will fetch an access token for a client application which will allow them to access the spotify API. It stores the access token in `client._spotifyToken` (although the developer can move it to where they want, the tokens expire and dont need to be stored persistently)

song/artist command will search the spotify API for a song/artist with a query given by the user, using the access token retrieved from the task.

### CODE REVIEW

**Comment by gc** [comment]
Example outputs of the 2 commands:
discord_2018-10-10_02-25-37
discord_2018-10-10_02-25-24

File: commands/Tools/artist.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	constructor(...args) {
+		super(...args, {
+			cooldown: 10,
+			description: 'Searches Spotify for an artist, and returns an embed.',
+			usage: '<query:str{1,100}>'
+		});
+	}
+
+	async run(msg, [query]) {
+		if (!this.client._spotifyToken) {
```
Comment (kyranet) [inline commands/Tools/artist.js]: I'm not completely convinced with this. This should be externalized as a constant like all other pieces that use an API key.

Comment (gc) [reply commands/Tools/artist.js]: its not a constant value though, it changes every hour?

Comment (kyranet) [reply commands/Tools/artist.js]: Oh right, you update it in tasks/getSpotifyToken. You should make a large JSDoc specifying that requirement.

File: tasks/getSpotifyToken.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+const { Task } = require('klasa');
+const fetch = require('node-fetch');
+
+// Login to https://developer.spotify.com/dashboard/applications
+// Create an app & get your clientID/secret, place them below
+// Run every 1 hour to refresh: this.client.schedule.create("getSpotifyToken", "*/60 * * * *")
```
Comment (kyranet) [inline tasks/getSpotifyToken.js]: Semicolon here

File: tasks/getSpotifyToken.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+			method: 'POST',
+			body: 'grant_type=client_credentials',
+			headers: {
+				Authorization: `Basic ${authorization}`,
+				'Content-Type': 'application/x-www-form-urlencoded'
+			}
+		})
+			.then(response => response.json())
+			.then(response => response.access_token)
+			.catch(console.error);
```
Comment (kyranet) [inline tasks/getSpotifyToken.js]: This should use a try/catch. And use `this.client.emit('wtf', error);` instead. The catch logs in an inconsistent style (no timestamp, no shard ID...) and makes accessToken be assigned to undefined, it should be assigned to null.

File: tasks/getSpotifyToken.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+	async init() {
+		this._getToken();
+	}
+
+	async _getToken() {
+		const accessToken = await fetch(`https://accounts.spotify.com/api/token`, {
+			method: 'POST',
+			body: 'grant_type=client_credentials',
+			headers: {
```
Comment (kyranet) [inline tasks/getSpotifyToken.js]: Headers could go outside the task to avoid re-creating it.

File: commands/Tools/song.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+					Accept: 'application/json',
+					'Content-Type': 'application/json',
+					Authorization: `Bearer ${this.client._spotifyToken}`
+				}
+			})
+			.then(response => response.json())
+			.then(response => response.tracks.items[0])
+			.catch(() => { throw 'There was an error. Please try again later.'; });
+
+		if (!song) throw "Couldn't find any songs with that name.";
```
Comment (kyranet) [inline commands/Tools/song.js]:
```javascript
if (song) return msg.sendMessage('...');
throw '...';
```

File: tasks/getSpotifyToken.js (hunk 2 above)
Comment (kyranet) [reply tasks/getSpotifyToken.js]: */60 is the same as 0

File: commands/Tools/artist.js (hunk 1 above)
Comment (KingDGrizzle) [inline commands/Tools/artist.js]: This shouldn't throw... It should try getting the task and running it, not tell you to `wait a bit`

Comment (gc) [reply commands/Tools/artist.js]: the task runs on startup, and every hour, if this statement has triggered then the request itself is probably not working and a wtf error should be emitted instead, as it should never be undefined

**Pull request merged**